bool heartbeat
bool estop_status